packages/rum-core/src/common/patching/event-target-patch.js

```diff
@@ -0,0 +1,185 @@
+/**
+ * MIT License
+ *
+ * Copyright (c) 2017-present, Elasticsearch BV
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+import {
+  SCHEDULE,
+  INVOKE,
+  ADD_EVENT_LISTENER_STR,
+  REMOVE_EVENT_LISTENER_STR,
+  EVENT_TARGET
+} from '../constants'
+
+import { apmSymbol } from './patch-utils'
+
+const eventTypes = ['click']
+
+const eventTypeSymbols = {}
+
+for (let i = 0; i < eventTypes.length; i++) {
+  const et = eventTypes[i]
+  eventTypeSymbols[et] = apmSymbol(et)
+}
+
+export function patchEventTarget(callback) {
+  if (!window.EventTarget) {
+    return
+  }
+
+  let proto = window.EventTarget.prototype
+  const nativeAddEventListener = proto[ADD_EVENT_LISTENER_STR]
+  const nativeRemoveEventListener = proto[REMOVE_EVENT_LISTENER_STR]
+
+  function findTaskIndex(existingTasks, eventType, listenerFn, capture) {
+    for (let i = 0; i < existingTasks.length; i++) {
+      const task = existingTasks[i]
+      if (
+        task.eventType === eventType &&
+        task.listenerFn === listenerFn &&
+        task.capture === capture
+      ) {
+        return i
+      }
+    }
+    return -1
+  }
+
+  function isCapture(options) {
```

Comment:
Seeing the below logic, Shouldn't something like this be enough? 
```
isCapture = (options && options.capture || options) === true;
```

Reply:
I've simplified the function a little bit. The above expression won't work if option is `{capture:1}`. but, In any case I prefer to be explicit and expand the logical expression to avoid errors.